MAUI/DataForm/Data-annotations.md

```diff
@@ -0,0 +1,475 @@
+
+## Display attribute
+
+<table>
+<tr>
+<th>Property</th>
+<th>Details</th>
+</tr>
+<tr>
+<td>
+{{'[Name] `Link`
+</td>
+<td>
+Specifies the label text.
+</td>
+</tr>
+<tr>
+<td>
+{{'[GroupName]`Link`
+</td>
+<td>
+Specifies the group name which groups the fields in the data form. Refer to  `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[ShortName] `Link`
+</td>
+<td>
+Specifies the label text. It takes higher priority than Name.
+</td>
+</tr>
+<tr>
+<td>
+{{'[AutoGenerateField]`Link`
+</td>
+<td>
+Specifies whether the field should be auto generated or not. Refer to `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[ResourceType]`Link`
+</td>
+<td>
+Specifies the Resources File (.Resx) which is used to localize the Display attribute of `Name`, `ShortName`, `GroupName` and `Prompt` values.
+</td>
+</tr>
+<tr>
+<td>
+{{'[Prompt]`Link`
+</td>
+<td>
+Specifies watermark text for the editor. Refer to `Link` for more details.
+</td>
+</tr>
+<tr>
+<td>
+{{'[Order] `Link`
+</td>
+<td>
+Specifies the order of field in the data form. Refer to `Link` for more details.
+</td>
+</tr>
+</table>
+
+{% tabs %}
+{% highlight Name attribute}
+
+    [Display(Name= "First Name")]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight GroupName attribute}
+
+    [Display(GroupName= "Address")]
+    public string DoorNO { get; set; }
+
+    [Display(GroupName = "Address")]
+    public string Street { get; set; }
+
+    [Display(GroupName = "Address")]
+    public string City { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight ShortName attribute}
+
+    [Display(ShortName= "First Name")]
+    public string Name { get; set; }
+
+    [Display(AutoGenerateField =false)]
+    public string Password { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight AutoGenerateField attribute}
+
+    [Display(AutoGenerateField =false)]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight ResourceType attribute}
+
+   -----------------------
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Prompt attribute}
+
+    [Display(Prompt = "Enter your name")]
+    public string Name { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+{% tabs %}
+{% highlight Order attribute}
+
+    [Display(Order =2)]
+    public string MiddleName { get; set; }
+
+    [Display(Order =1)]
+    public string LastName { get; set; }
+
+    [Display(Order =0)]
+    public string FirstName { get; set; }
+
+{% endhighlight %}
+{% endtabs %}
+
+
+
+## Validation attribute
+
+<table>
+<tr>
+<th>
+Property
+</th>
+<th>
+Details
+</th>
+</tr>
+<tr>
+<td>
+{{'[MinLength](https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.minlengthattribute?view=net-7.0)'| markdownify }}
+</td>
+<td>
+Specifies the required minimum length. Refer to `Link` for more details.
```

Comment:
Specifies the required minimum input string length.

Reply:
new content added